File: src/main/java/com/back/nbe9112team06/domain/timetable/repository/TimeTableRepository.java
package com.back.nbe9112team06.domain.timetable.repository;

import com.back.nbe9112team06.domain.timetable.entity.TimeTable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TimeTableRepository extends JpaRepository<TimeTable, Integer> {
    List<TimeTable> findByMeeting_Id(Integer meetingId);
}

File: src/main/kotlin/com/back/nbe9112team06/domain/timetable/repository/TimeTableRepository.kt
package com.back.nbe9112team06.domain.timetable.repository

import com.back.nbe9112team06.domain.timetable.entity.TimeTable
import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.data.jpa.repository.Query

interface TimeTableRepository : JpaRepository<TimeTable, Int> {
    fun findByMeetingId(meetingId: Int): MutableList<TimeTable>

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    @Query(
        """
    select tt
    from TimeTable tt
    where tt.meeting.id = :meetingId
    """
    )
    fun findByMeetingIdForUpdate(meetingId: Int): TimeTable?
}
